lib/settings/settings_screen.dart

```diff
@@ -39,37 +47,195 @@ class SettingsScreen extends StatefulWidget {
+  OrionConfig config = OrionConfig();
+  Logger logger = Logger('Settings');
+  late bool needsRestart;
+  final GlobalKey<WifiScreenState> _wifiScreenKey =
+      GlobalKey<WifiScreenState>();
+  final ValueNotifier<bool> isConnected = ValueNotifier(false);
+  late Future<void> _wifiScreenFuture;
+
+  @override
+  void initState() {
+    super.initState();
+    needsRestart = config.getFlag('needsRestart', category: 'internal');
+    _wifiScreenFuture = _initializeWifiScreen();
+  }
+
+  Future<void> _initializeWifiScreen() async {
+    final bool initialConnectionStatus = await _checkInitialConnectionStatus();
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      if (mounted) {
+        isConnected.value = initialConnectionStatus;
+      }
+    });
+  }
+
+  Future<bool> _checkInitialConnectionStatus() async {
+    try {
+      final result = await Process.run('ping', ['-c', '1', 'google.com']);
+      return result.exitCode == 0;
+    } catch (e) {
+      logger.severe('Failed to check initial connection status: $e');
+      return false;
+    }
+  }
 
   void _onItemTapped(int index) {
     setState(() {
       _selectedIndex = index;
     });
   }
 
-  @override
-  void initState() {
-    super.initState();
+  void launchConfirmationDialog(bool closeSettings) async {
+    if (!needsRestart || !Platform.isLinux) return;
+    bool shouldRestart = await showDialog(
+      barrierDismissible: false,
+      context: context,
+      builder: (BuildContext context) {
+        return AlertDialog(
+          title: const Text('Finalize Settings'),
+          content: const Text(
+              'The Touch Interface needs to restart to apply changes.\nDo you want to restart now or later?'),
+          actions: [
+            TextButton(
+              onPressed: () {
+                Navigator.of(context).pop(false);
+                if (closeSettings) Navigator.of(context).pop();
+              },
+              child: const Text('Later', style: TextStyle(fontSize: 20)),
+            ),
+            const SizedBox(width: 20),
+            TextButton(
+              onPressed: () {
+                Navigator.of(context).pop(true);
+              },
+              child: const Text('Now', style: TextStyle(fontSize: 20)),
+            ),
+          ],
+        );
+      },
+    );
+
+    if (shouldRestart) {
+      logger.info('Restarting Orion');
+      config.setFlag('needsRestart', false, category: 'internal');
+      setState(() {
+        needsRestart = false;
+      });
+      restartOrion();
+    }
+  }
+
+  Future<void> launchDisconnectDialog() async {
+    bool shouldDisconnect = await showDialog(
+      barrierDismissible: false,
+      context: context,
+      builder: (BuildContext context) {
+        return AlertDialog(
+          title: const Text('Disconnect from WiFi'),
+          content: const Text(
+              'Do you want to disconnect from the current WiFi network?\nThis may cause any ongoing print jobs to fail.'),
```

Comment:
Is this true? I wouldn't expect so??

Reply:
That warning is under consideration that the user might be connected to a remote machine, so disconnecting could have unforseen consequences.

Reply:
But shouldn't odyssey continue running as normal? I wouldn't expect a UI disconnect to do anything, remote or not.

Reply:
Looks good